### DISCUSSION TITLE
Extracting 2D drawings from iModel yields empty results

### DISCUSSION THREAD

**Discussion by salah-walid** [iTwin/itwinjs-core#5573] [Q&A]
Using Synchro control and ITwin 3.7.0, and following [this sample](https://www.itwinjs.org/sandboxes/iTwinPlatform/2d%20Viewer), i tried to extract drawings and sheets from my iModel it yields empty results.

I have tried with revit, navisworks and synchro files and they all yield empty results when querying 2D data like this

 ```Typescript
 await imodel.models.queryProps({ from: "BisCore.GeometricModel2d" });
```
It return an empty list, the only time it somewhat returned a result is when i used a DWG file but it returned an empty drawing.
Am i missing something in the upload or there are only some type of files that can have 2D data extracted ?

**Comment by pratikshan85** [comment]
@salah-walid Can you please give us the input file to create your iModel internally so that we can investigate the issue in more details? 
Also, can you give us more details about your code snippet?

**Reply by salah-walid** [reply]
[Here is the link to the revit file i used](https://drive.google.com/file/d/1G2xWy5_3er9tUFjlmVSd1cCn7Hv9pjlX/view?usp=sharing)

For the code here is the frontstage class:
```Typescript
export class MultiViewContentGroupProvider extends ContentGroupProvider {
  public async provideContentGroup(
    _props: FrontstageProps
  ): Promise<ContentGroup> {
    //.....................................................................
    const result = await ViewerOnly2dApi.getInitial2DModel(iModel!)
    let viewState2D = await ViewerOnly2dApi.createDefaultViewFor2dModel(
      iModel!,
      result
    )

    return new ContentGroup({
      id: "custom-stage-content-id",
      layout: {
        id: "TwoHalvesHorizontal",
        verticalSplit: {
          id: "TwoHalvesHorizontalSplit",
          percentage: 0.7,
          right: 0,
          left: 1,
        },
      },
      contents: [
        {
          id: "2DContent",
          classId: IModelViewportControl,
          applicationData: {
            viewState: viewState2D,
            iModelConnection: iModel,
            supports: ["2dModels"],
          },
        },
        {
          id: "primaryContent",
          classId: IModelViewportControl,
          applicationData: {
            isPrimaryView: true,
            supports: ["viewIdSelection", "3dModels"],
            viewState,
            iModelConnection: iModel,
          },
        },
      ],
    })
  }
}
```

ViewerOnly2dApi is the same one used [here](ViewerOnly2dApi)

**Reply by navneetk897** [reply]
@salah-walid I have created an imodel with the file that you provided in the previous response and ran some manual query like 
```SELECT * FROM bis.GeometricModel2d``` or ```select * from bis.GeometricElement2d``` and the result was empty which means your file doesn't contain any 2d sheets.

![image](https://github.com/iTwin/itwinjs-core/assets/67821826/0a1f34d6-2e99-493f-a963-1ad6a9dccaa0)
![image](https://github.com/iTwin/itwinjs-core/assets/67821826/ff8acbbb-f409-4f91-a627-306e9e1a8785)

In the viewer also the option to toggle 2d view is unclickable.

![viewError](https://github.com/iTwin/itwinjs-core/assets/67821826/aba60820-74a3-4607-9c50-ee9f6087712c)

So, You first need to check if your file contain 2d sheets or not, if not then you need to create one.

**Reply by salah-walid** [reply]
Opening the file it clearly has sheets
![image](https://github.com/iTwin/itwinjs-core/assets/29980658/79dcf788-0467-42fa-a246-771e5289c823)

I suspect that the revit ITwin interpreter doesn't translate revit sheets, i have tried DWG files but it gives me empty sheets in ITwin

**Reply by navneetk897** [reply]
@salah-walid we are looking into this issue. we will get back to you soon.

**Comment by pratikshan85** [comment]
@salah-walid We had a detailed discussion with our dev team regarding the issue and internally we need to set a flag to get the 2D sheet elements. Can you please provide us the ProjectId you are using for this iModel so that we can internally set the flag for this project to get the desired output for you.

**Note : Currently the output you are getting in the iModel this is the default bhavior.**

**Reply by salah-walid** [reply]
Thank you for your feedback, but this solution is not scalable for the long run with multiple new projects that we have to create, is the ITwin team looking to fixing this issue? 
Can we have the ability to modify this flag in synchro control ?

**Reply by pmconne** [reply]
You're right @salah-walid, that would not be scalable. The iTwin platform's synchronization APIs already support configuration via a provided manifest, which can override this behavior. We have asked the Synchro Control folks to consider exposing the option to include sheets+drawings when importing from Revit. Please feel free to [raise an issue](https://bentleysystems.service-now.com/csp) with them.

In the meanwhile, we can manually set the flag on a per-project basis if provided with the project Id(s).
